services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/MetersSpec.groovy

```diff
@@ -0,0 +1,219 @@
+        def sw = topology.getActiveSwitches().find { it.ofVersion == "OF_13" }
+
+        when: "Get default meters from the switch"
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        assert defaultMeters
+
+        and: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+
+        then: "New meters should appear after flow setup"
+        def newMeters = northboundService.getAllMeters(sw.dpId)
+        def newMeterEntries = newMeters.meterEntries.findAll { !defaultMeters.meterEntries.contains(it) }
+        newMeterEntries.size() == 2
+
+        and: "All new meters should have KBPS, BURST and STATS flags installed"
+        newMeterEntries.every { it.flags.sort().equals(["KBPS", "BURST", "STATS"].sort()) }
+
+        and: "All new meters rate should be equal to flow's rate"
+        newMeterEntries*.rate.every { it == flow.maximumBandwidth }
+
+        when: "Delete a flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "New meters should disappear from the switch"
+        def newestMeters = northboundService.getAllMeters(sw.dpId)
+        newestMeters.meterEntries.containsAll(defaultMeters.meterEntries)
+        newestMeters.meterEntries.size() == defaultMeters.meterEntries.size()
+    }
+
+    @Unroll
+    @Ignore("Until the fuzzy logic of Noviflow bursting algorithm is clear")
+    def "Meter burst size should be #burstSize kbps if the flow rate is #flowRate kbps"() {
+        setup: "A flow with #flowRate kbps bandwidth is created on OpenFlow 1.3 compatible switch"
+        def sw = getNonCentecSwitches()[0]
+        def defaultMeters = northboundService.getAllMeters(sw.dpId)
+        def flowPayload = flowHelper.singleSwitchFlow(sw)
+        flowPayload.setMaximumBandwidth(100)
+        def flow = flowHelper.addFlow(flowPayload)
+
+        when: "Update flow bandwidth to #flowRate kbps"
+        flow = northboundService.getFlow(flow.id)
+        flow.setMaximumBandwidth(flowRate)
+        northboundService.updateFlow(flow.id, flow)
+        sleep(Constants.RULES_INSTALLATION_TIME * 1000) //TODO: add helper method to actually monitor this
+
+        then: "New meters should be installed on the switch"
+        def newMeters = northboundService.getAllMeters(sw.dpId).meterEntries.findAll {
+            !defaultMeters.meterEntries.contains(it)
+        }
+        newMeters.size() == 2
+
+        and: "New meters rate should be equal to flow bandwidth"
+        newMeters*.rate.every { it == flowRate }
+
+        and: "New meters burst size should be between 100.5% and 105% of the flow's rate"
+        newMeters*.burstSize.every { (it >= flowRate * NOVI_BURST_MIN) && (it <= flowRate * NOVI_BURST_MAX) }
+
+        cleanup: "Delete the flow"
+        northboundService.deleteFlow(flow.id)
+
+        where:
+        flowRate << [150, 1000, 1024, 5120, 10240, 2480, 960000]
+    }
+
+    def "Flow burst should not exceed 1024 kbps on Centec switches"() {
```

Comment:
Not quite accurate description. It should be equal to 1024 (not `not exceed`)

Reply:
Fixed the description